core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -845,10 +856,25 @@ oppia.factory('explorationStatesService', [
       }
     };
 
+    var convertToBackend = function(propertyValue) {
+      var propToBackend = [];
+      for (var i = 0; i < propertyValue.length; i++) {
+        propToBackend.push(propertyValue[i].toBackendDict());
+      }
+      return propToBackend;
+    };
+
     // TODO(sll): Add unit tests for all get/save methods.
     return {
-      init: function(value) {
-        _states = angular.copy(value);
+      init: function(states) {
+        _states = {};
+        for (var stateName in states) {
+          if (states.hasOwnProperty(stateName)) {
```

Comment:
Um, obviously it does have that property. I mean, you're iterating over the properties in `states` in the previous line...